deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/ErrorCodeDescription.java

```diff
@@ -0,0 +1,193 @@
+package com.microsoft.azure.sdk.iot.deps.serializer;
+
+/**
+ * See https://docs.microsoft.com/en-us/rest/api/iothub/common-error-codes for additional details
+ */
+public enum ErrorCodeDescription
+{
+    // These descriptions belong to fully qualified status codes, such as 404001
+    UnclassifiedErrorCode,
+    InvalidProtocolVersion,
+    DeviceInvalidResultCount,
+    InvalidOperation,
+    ArgumentInvalid,
+    ArgumentNull,
+    IotHubFormatError,
+    DeviceStorageEntitySerializationError,
+    BlobContainerValidationError,
+    ImportWarningExistsError,
+    InvalidSchemaVersion,
+    DeviceDefinedMultipleTimes,
+    DeserializationError,
+    BulkRegistryOperationFailure,
+    CannotRegisterModuleToModule,
+    IotHubNotFound,
+    IotHubUnauthorizedAccess,
+    IotHubUnauthorized,
+    IotHubSuspended,
+    IotHubQuotaExceeded,
+    JobQuotaExceeded,
+    DeviceMaximumQueueDepthExceeded,
+    IotHubMaxCbsTokenExceeded,
+    DeviceNotFound,
+    JobNotFound,
+    PartitionNotFound,
+    ModuleNotFound,
+    DeviceAlreadyExists,
+    ModuleAlreadyExistsOnDevice,
+    DeviceMessageLockLost,
+    MessageTooLarge,
+    TooManyDevices,
+    TooManyModulesOnDevice,
+    ThrottleBacklogLimitExceeded,
+    InvalidThrottleParameter,
+    ServerError,
+    JobCancelled,
+    ConnectionForcefullyClosedOnNewConnection,
+    DeviceNotOnline,
+    DeviceConnectionClosedRemotely,
+
+    // These descriptions belong to 3 digit http response codes such as 404.
+    BadFormat,
+    Unauthorized,
+    Forbidden,
+    NotFound,
+    Conflict,
+
+    /**
+     * Represents status code for 429 and for status code 429001
+     */
+    PreconditionFailed,
+
+    RequestEntityTooLarge,
+
+    /**
+     * Represents status code for 429 and 429001
+     */
+    ThrottlingException,
+    InternalServerError,
+
+    /**
+     * Defined for status code 503 and 503001
+     */
+    ServiceUnavailable;
```

Comment:
Both of these codes map to ServiceUnavailable?

Reply:
Yep, weirdly

Reply:
Registry manager ops gets 503, whereas a device method invocation likely gets a 503001 based on my research so far. In both cases, though, the service is telling the SDK that the service is unavailable